app/models/concerns/trackable_job.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+##
+module TrackableJob
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+  def self.included(base)
+    base.class_eval do
+      before_enqueue do |job|
+        tracker = job_tracker(job.job_id, job.arguments.first)
+        tracker.update(status: 'ENQUEUED')
```

Comment:
Good idea. I've added a second commit that deals with making these constants and validating the model with them.